File: Depth First Search/iterative.js
class Node {
    constructor(name) {
      this.name = name;
      this.children = [];
    }
  
    addChild(name) {
      this.children.push(new Node(name));
      return this;
    }
  
    depthFirstSearch(array) {
      // Write your code here.
      if(!this) return array
      let stack=[this]
      while (stack.length){
        const currNode = stack.pop()
        array.push(currNode?.name)
        for (let i=currNode?.children?.length-1;i>=0;i--){
          currNode.children[i] && stack.push(currNode.children[i])
        }
      }
      return array
    }
  }

File: Depth First Search/iterative.py
class Node:
    def __init__(self, name):
        self.children = []
        self.name = name

    def addChild(self, name):
        self.children.append(Node(name))
        return self

    def depthFirstSearch(self, array):
        # Write your code here.
        if self is None: return array
        stack=[self]
        while len(stack):
            currNode= stack.pop()
            array.append(currNode.name)
            for child in reversed(currNode.children):
                stack.append(child)
        return array
